src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +98,725 @@ ActorInit En_Dt_InitVars = {
+    EnDt_UpdateHeadRotate(this);
 
-extern UNK_TYPE D_0600112C;
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
+        Message_CloseTextbox(play);
+        if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_SPOKE_WITH_MAYOR_ON_NIGHT_3)) {
+            this->textIdIndex = 25;
+            Message_ContinueTextbox(play, sTextIds[this->textIdIndex]);
+            SET_WEEKEVENTREG(WEEKEVENTREG_SPOKE_WITH_MAYOR_ON_NIGHT_3);
+        } else {
+            EnDt_SetupFinalNightState(this, play);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/EnDt_Init.s")
+void EnDt_Update(Actor* thisx, PlayState* play) {
+    EnDt* this = THIS;
+    ColliderCylinder* collider;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/EnDt_Destroy.s")
+    SkelAnime_Update(&this->skelAnime);
+    Actor_SetScale(&this->actor, 0.01f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9C74.s")
+    if ((this->state != 4) && (this->state != 5) && (gSaveContext.save.day == 3) && gSaveContext.save.isNight) {
+        EnDt_SetupFinalNightState(this, play);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9CE8.s")
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_RESOLVED_MAYOR_MEETING) &&
+        (gSaveContext.save.day != 3 || (gSaveContext.save.day == 3 && !gSaveContext.save.isNight))) {
+        Audio_PlaySequenceAtPos(SEQ_PLAYER_BGM_SUB, &gSfxDefaultPos, NA_BGM_MAYORS_OFFICE, 1000.0f);
+        Actor_PlaySfx(&this->actor, NA_SE_EV_CROWD - SFX_FLAG);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9D9C.s")
+    DECR(this->blinkTimer);
+    DECR(this->timer);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9DF8.s")
+    if (this->meetingFinished) {
+        EnDt_UpdateHeadRotate(this);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9E94.s")
+    // Blinking update
+    if (!this->disableBlinking && (this->blinkTimer == 0)) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex >= EN_DT_EYE_TEXTURE_LOOK_DOWN) {
+            this->eyeTexIndex = EN_DT_EYE_TEXTURE_SHOCK;
+            this->blinkTimer = (s32)Rand_ZeroFloat(60.0f) + 20;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9EF8.s")
+    this->actionFunc(this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BEA088.s")
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Math_SmoothStepToS(&this->headRotValue.y, this->headRotTarget.y, 1, 3000, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BEA254.s")
+    Actor_SetFocus(&this->actor, 60.0f);
+    Actor_MoveWithGravity(&this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BEA394.s")
+    collider = &this->collider;
+    Collider_UpdateCylinder(&this->actor, collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &collider->base);
```

Comment:
```suggestion
    Collider_UpdateCylinder(&this->actor, &this->collider);
    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
```
Is this temp actually needed. If it causes the stack to be too small just add an `s32 pad`. Note this pad would actually be real, and if you curious ping me in discord (anon58) and I can explain it.

Reply:
You're right, it is not actually needed with the padding.